src/annotator/util/test/extract-annotation-query-test.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+var annotationIds = require('../extract-annotation-query');
+
+describe('annotation queries', function () {
+  var annotation = annotationIds.extractAnnotationQuery('http://localhost:3000#annotations:alphanum3ric_-only');
+  var queryVarA = annotationIds.extractAnnotationQuery('http://localhost:3000#annotations:q:user:USERNAME');
+  var queryVarB = annotationIds.extractAnnotationQuery('http://localhost:3000#annotations:QuerY:user:USERNAME');
+  var invalid = annotationIds.extractAnnotationQuery('http://localhost:3000#annotations:\"TRYINGTOGETIN\";EVILSCRIPT()');
```

Comment:
A general comment about these tests is that it is easier to follow if the preparation/arrangement is part of the test case instead of being a separate step that happens in `beforeEach`. The exception is if you're using a particular fixture across several test cases. We generally try to follow the 'Arrange Act Assert' formula for structuring individual test cases.